enshrine-esdt-safe/src/lib.rs

```diff
@@ -45,6 +45,13 @@ pub trait EnshrineEsdtSafe:
         self.header_verifier_address().set(&header_verifier_address);
     }
 
+    #[only_owner]
+    #[endpoint(setWegldTicker)]
+    fn set_wegld_ticker(&self, wegld_ticker: ManagedBuffer) {
```

Comment:
I believe it's better to have the identifier in `init` as argument `Optional<ManagedBuffer>`.
It is only required for main chain contract, so you can do a check like this:
```
if is_sovereign_chain == false {
   //wegld_identifier should have value -> otherwise return error
}
```

Reply:
we can just use a `require!` to check for the `is_sovereign_chain` value to be true inside this function, this will also mean we don't have to use a `match` operator

Reply:
If you move this in `init` you can't use the require.